internal/builtins/pkg_context.go

```diff
@@ -118,9 +130,16 @@ func pkgContextResource(kf *yaml.RNode) (*yaml.RNode, error) {
 			return nil, err
 		}
 	}
-	cm.SetDataMap(map[string]string{
-		"name": kf.GetName(),
-	})
+	data := map[string]string{
+		"name": kptfile.GetName(),
+	}
+	if packageConfig != nil {
+		if packageConfig.PackagePath != "" {
+			data[ConfigKeyPackagePath] = packageConfig.PackagePath
+		}
+	}
```

Comment:
I am guessing this conditional will ensure package-path is not added for `kpt CLI` based workflows ?

Reply:
I believe so, correct.  It'll set it only if we have something to set.

Based on the discussion above, I sort of feel like we shouldn't be persisting either package-path or package-name into the package-context.yaml.  Rather we should be generating it on the fly every time; then we don't have a conflict between directory name / directory path vs package-name/package-path.

The exception to that would be if we wanted the _original_ name, so that if we generated something immutable it would be preserved.  However, we do have other answers there, like trying to capture the name in the parameters in the kpt function pipeline or similar.

Sounds like we need to think more about the challenges here, but that we should merge this so we can gain some experience with how it behaves in practice.